### ISSUE TITLE
Build failure: `ledger`

### ISSUE THREAD

**Issue by dtgoitia** [NixOS/nixpkgs#368785]
## Steps To Reproduce

1. I tried to install `ledger` using the following `flake.nix` file in Arch Linux:

<details>
<summary>flake.nix</summary>

```
{
  description = "My user-level packages for dtg@beltz";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    nixpkgs-popcorntime.url = "github:NixOS/nixpkgs/c3997c9525ed62c4cf4195e5ca27440a083476d5";
    nixvim.url = "github:nix-community/nixvim";
  };

  outputs = {
    self,
    nixpkgs,
    nixpkgs-popcorntime,
    nixvim,
  }: let
    system = "x86_64-linux";

    pkgs = nixpkgs.legacyPackages.${system};
    popcorntime-pkgs = nixpkgs-popcorntime.legacyPackages.${system};

    # I don't understand why this code evaluates to the packages themselves... I would have
    # expected to need to access `.legacyPackages` at some point, but it's not needed
    unfree-pkgs = import nixpkgs {
      inherit system;
      config.allowUnfree = true;
    };

    nixvimLib = nixvim.lib.${system};
    nixvimForSystem = nixvim.legacyPackages.${system};
    nixvimModule = {
      inherit pkgs;
      module = import ./config/nvim; # import the module directly
      # You can use `extraSpecialArgs` to pass additional arguments to your module files
      extraSpecialArgs = {
        # inherit (inputs) foo;
      };
    };
    nvim = nixvimForSystem.makeNixvimWithModule nixvimModule;
  in {
    #
    # `buildEnv` creates a single environment to manage many packages as a single
    # one: it creates one derivation that contains symlinks to the other packages
    packages.${system}.default = pkgs.buildEnv {
      name = "my-declaratively-installed-packages";
      paths = [
        nvim

        pkgs.alejandra
        pkgs.jq
        pkgs.jump
        pkgs.keepassxc
        pkgs.ledger
        pkgs.nnn
        pkgs.speedtest-cli
        pkgs.tree
        pkgs.vlc
        pkgs.vscodium
        pkgs.yq
        #
        # not-free
        unfree-pkgs.ngrok
        #
        # pinned
        popcorntime-pkgs.popcorntime
        #
        # occasionally used
        # pkgs.dbeaver-bin # works with nixpkgs:634fd46801442d760e09493a794c4f15db2d0cbb
        # pkgs.inkscape # works with nixpkgs:634fd46801442d760e09493a794c4f15db2d0cbb
        # pkgs.krita
        # pkgs.xournalpp
      ];
    };
  };
}
```
</details>

## Build log

<details>

<summary>Build Log</summary>

```
error: builder for '/nix/store/nnpqp3cd7w0myp4hhs7zcs3jp20cnjvy-ledger-3.3.2.drv' failed with exit code 2;
       last 25 log lines:
       > /build/source/build/system.hh:147:10: fatal error: boost/filesystem/convenience.hpp: No such file or directory
       >   147 | #include <boost/filesystem/convenience.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [src/CMakeFiles/libledger.dir/build.make:104: src/CMakeFiles/libledger.dir/csv.cc.o] Error 1
       > In file included from /build/source/src/emacs.cc:32:
       > /build/source/build/system.hh:147:10: fatal error: boost/filesystem/convenience.hpp: No such file or directory
       >   147 | #include <boost/filesystem/convenience.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [src/CMakeFiles/libledger.dir/build.make:146: src/CMakeFiles/libledger.dir/emacs.cc.o] Error 1
       > In file included from /build/source/src/generate.cc:32:
       > /build/source/build/system.hh:147:10: fatal error: boost/filesystem/convenience.hpp: No such file or directory
       >   147 | #include <boost/filesystem/convenience.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [src/CMakeFiles/libledger.dir/build.make:90: src/CMakeFiles/libledger.dir/generate.cc.o] Error 1
       > In file included from /build/source/src/ptree.cc:32:
       > /build/source/build/system.hh:147:10: fatal error: boost/filesystem/convenience.hpp: No such file or directory
       >   147 | #include <boost/filesystem/convenience.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [src/CMakeFiles/libledger.dir/build.make:160: src/CMakeFiles/libledger.dir/ptree.cc.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:209: src/CMakeFiles/libledger.dir/all] Error 2
       > make: *** [Makefile:166: all] Error 2
       For full logs, run 'nix-store -l /nix/store/nnpqp3cd7w0myp4hhs7zcs3jp20cnjvy-ledger-3.3.2.drv'.
error: 1 dependencies of derivation '/nix/store/hcqw75d5vi1fhiiy2i28dz95sknmzx56-my-declaratively-installed-packages.drv' failed to build
```

</details>

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.67-1-lts, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/home/dtg/.nix-defexpr/channels/nixpkgs`

## Notify maintainers

@jwiegley

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by benmezger** [comment]
I can confirm the issue on my side too.

**Comment by acristoffers** [comment]
I missed this issue and opened a duplicate, sorry. I can confirm on nixos unstable as well.

**Comment by davisrichard437** [comment]
I can likewise confirm on nixos-unstable. It does seem thought that this was fixed in #359441, so it should land there in a few days.

In the meantime, if you need ledger, you can run the following:

    $ nix shell github:nixos/nixpkgs/master#ledger

It will have to compile once but then it should be cached and run just fine.

**Comment by SigmaSquadron** [comment]
The fix for `ledger` has reached `nixos-unstable`. Please update your systems.
